bindings/ffi/src/lib.rs

```diff
@@ -105,13 +163,69 @@ pub extern "C" fn regorus_result_drop(r: RegorusResult) {
         }
     }
 }
+// Extension function to invoke callback
+fn invoke_callback(mut params: Vec<::regorus::Value>) -> Result<::regorus::Value> {
+    if params.len() != 2 {
+        bail!("invoke requires exactly two parameters: function_name and payload");
+    }
+    
+    let function_name = match params.remove(0) {
+        ::regorus::Value::String(s) => s.as_ref().to_string(),
+        _ => bail!("function_name must be a string"),
+    };
+    
+    // Serialize the payload to JSON
+    let payload = serde_json::to_string(&params.remove(0))
+        .map_err(|e| anyhow!("Failed to serialize payload: {}", e))?;
+    
+    // Look up the callback function using thread_local storage
+    let callback_option = CALLBACK_MAP.with(|callbacks| {
+        callbacks.borrow().get(&function_name).cloned()
+    });
+    
+    let (callback_fn, context) = match callback_option {
+        Some(cb) => cb,
+        None => bail!("No callback registered with name: {}", function_name),
+    };
+    
+    // Convert payload to C string
+    let payload_c_str = match CString::new(payload) {
+        Ok(cs) => cs,
+        Err(_) => bail!("Failed to convert payload to C string"),
+    };
+    
+    // Call the orchestration function
+    let result_ptr = callback_fn(payload_c_str.as_ptr(), context);
+    
+    if result_ptr.is_null() {
+        return Ok(::regorus::Value::Null);
+    }
+    
+    // Convert the result back to a Value
+    let result_str = unsafe {
+        let result = CStr::from_ptr(result_ptr)
+            .to_str()
+            .map_err(|e| anyhow!("Invalid UTF-8 in callback result: {}", e))?
+            .to_string();
+            
+        // Free the memory allocated by C#
+        let _ = CString::from_raw(result_ptr as *mut c_char);
+        
+        result
+    };
+    
+    // Convert result string to Value
+    ::regorus::Value::from_json_str(&result_str)
+        .map_err(|e| anyhow!("Failed to parse callback result as JSON: {}", e))
+}
 
 #[no_mangle]
 /// Construct a new Engine
 ///
 /// See https://docs.rs/regorus/latest/regorus/struct.Engine.html
 pub extern "C" fn regorus_engine_new() -> *mut RegorusEngine {
-    let engine = ::regorus::Engine::new();
+    let mut engine = ::regorus::Engine::new();
+    let _ = engine.add_extension("invoke".to_string(), 2, Box::new(invoke_callback));
```

Comment:
To enable writing `function(args)` instead of `invoke("function", args)` in the rego policy, we could change the behavior of the interpreter so that it looks at the table of foreign language extensions if no function exists for a given name.

On the other hand, `invoke` does make it clear to the reader of the policy that a foreign language function is being used.

Reply:
I think `invoke()` would be a cleaner option. Although we can check that it is possible to to trigger invoke from another extension, so that user can optionally use `add_extension("function"...)` that will `invoke("function"...)`